test/unit/search/fm_index_cursor/bi_fm_index_cursor_collection_test.cpp

```diff
@@ -7,5 +7,47 @@
 
 #include "bi_fm_index_cursor_collection_test_template.hpp"
 
+#include <seqan3/range/views/char_to.hpp>
+#include <seqan3/range/views/to.hpp>
+
+#include <string_view>
+
+// generic type
+template <typename index_cursor_t>
+struct bi_fm_index_cursor_collection_test : public ::testing::Test
+{
+    using index_type = typename index_cursor_t::index_type;
+    using alphabet_type = typename index_type::alphabet_type;
+    using text_type = std::vector<alphabet_type>;
+
+    static constexpr auto convert = seqan3::views::char_to<alphabet_type> | seqan3::views::to<text_type>;
+
+    text_type text{convert(std::string_view{"ACGGTAGGACGTAGC"})};
+    text_type text1{convert(std::string_view{"AACGATCGGA"})};
+    text_type text2{convert(std::string_view{"TGCTACGATCC"})};
+    text_type text3 = seqan3::views::slice(text, 0, 11) | seqan3::views::to<text_type>; // "ACGGTAGGACG"
+    text_type text4 = seqan3::views::slice(text, 0, 14) | seqan3::views::to<text_type>; // "ACGGTAGGACGTAG"
+
+    std::vector<text_type> text_col1{text1, text1};
+    std::vector<text_type> text_col2{text3, text2};
+    std::vector<text_type> text_col3{text4, text2};
+    std::vector<text_type> text_col4{text, text2};
+
+    std::vector<text_type> rev_text1 = text_col1 | seqan3::views::deep{std::views::reverse}
```

Comment:
please also include `<seqan3/std/ranges>`, persist and also the alphabets that you are using.
Right now it works because of some dependencies. But in case they get removed or something than tests might stop compiling because they are not getting the things they need.